### PR TITLE
Sync .packages file as a normal input file.

### PR DESCRIPTION
**Pull request by aam**
Currently `.packages` file is treated specially as DevFSStringContent. We need to treat it as a regular file because from dart2 frontend perspective it's just a normal input file that should be invalidated if it changes. Syncing of `.packages` to the device for `no-preview-dart-2` flow also works as expected when we treat it as a normal file.

### CODE REVIEW

File: packages/flutter_tools/lib/src/devfs.dart [hunk 1]
```diff
@@ -581,8 +581,9 @@ class DevFS {
     }
     assert((file is Link) || (file is File));
     if (ignoreDotFiles && fs.path.basename(file.path).startsWith('.')) {
-      // Skip dot files.
-      return true;
+      // Skip dot files, but not the '.packages' file (even though in dart1
+      // mode devfs['.packages'] will be overwritten with synthesized string content).
+      return fs.path.basename(file.path) != '.packages';
```
Comment (a-siva) [inline packages/flutter_tools/lib/src/devfs.dart]: fs.path.basename(file.path) is called twice, does it make sense to hoist that into a local variable?

File: packages/flutter_tools/lib/src/devfs.dart [hunk 2]
```diff
@@ -722,6 +723,14 @@ class DevFS {
       }
     }
     if (sb != null) {
+      if (previewDart2) {
```
Comment (a-siva) [inline packages/flutter_tools/lib/src/devfs.dart]: Why not
if (!previewDart2 && (sb != null)) {
  .....
}

File: packages/flutter_tools/lib/src/devfs.dart [hunk 3]
```diff
@@ -424,7 +424,7 @@ class DevFS {
                          fileFilter: fileFilter);
     if (fs.isFileSync(_packagesFilePath)) {
       printTrace('Scanning package files');
-      await _scanPackages(fileFilter);
+      await _scanPackages(fileFilter, generator != null);
```
Comment (a-siva) [inline packages/flutter_tools/lib/src/devfs.dart]: Can you add a comment here that we detect we are in previewDart2 mode by generator (should have been named compiler?? is present) or have a local
bool previewDart2 = (generator != null);
to make that more obvious.

**Review by a-siva — APPROVED** [review]

File: packages/flutter_tools/lib/src/devfs.dart (hunk 2 above)
Comment (aam) [reply packages/flutter_tools/lib/src/devfs.dart]: Refactored slightly.

File: packages/flutter_tools/lib/src/devfs.dart (hunk 3 above)
Comment (aam) [reply packages/flutter_tools/lib/src/devfs.dart]: Done

File: packages/flutter_tools/lib/src/devfs.dart (hunk 1 above)
Comment (aam) [reply packages/flutter_tools/lib/src/devfs.dart]: Done

**Pull request merged**